### PR TITLE
feat: add api get user tag

### PR DESCRIPTION
**Pull request by NguyenHuy1812**
**What does this PR do?**

-   [x] Add api get user tag 
-   [x] Add api upsert user tag
Screenshot 2023-04-10 at 14 06 58

### CODE REVIEW

File: migrations/schemas/20230410054010-add-table-user-tags.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
+-- +migrate Up
+CREATE TABLE IF NOT EXISTS user_tags
+(
+    id UUID PRIMARY KEY NOT NULL DEFAULT uuid_generate_v4(),
```
Comment (vhbien000) [inline migrations/schemas/20230410054010-add-table-user-tags.sql:5]: should we use serial (bigint or string if using snowflake) for compatible with current user id as I remember. UUID is usually used to take advantage of secure and privacy, but not good for indexing, searching, and human-readable.

File: migrations/schemas/20230410054010-add-table-user-tags.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    guild_id integer,
```
Comment (vhbien000) [inline migrations/schemas/20230410054010-add-table-user-tags.sql:6]: guild_id and user_id of discord is snowflake so I think we should store them in `string`

File: pkg/handler/community/community.go [hunk 3]
```diff
@@ -823,3 +824,58 @@ func (h *Handler) UpdateAdById(c *gin.Context) {
...
+// @Param       req  query request.UpsertUserTag true "Upsert user tag request"
+// @Success     200 {object} response.ResponseMessage
+// @Router      /community/tagme [post]
+func (h *Handler) UpsertUserTag(c *gin.Context) {
+	req := request.UpsertUserTag{}
+	if err := c.BindJSON(&req); err != nil {
+		h.log.Error(err, "[handler.UpsertUserTag] BindJSON() failed")
+		c.JSON(http.StatusBadRequest, response.CreateResponse[any](nil, nil, err, nil))
+		return
+	}
```
Comment (vhbien000) [inline pkg/handler/community/community.go:843]: should add line break here

File: pkg/handler/community/community.go [hunk 4]
```diff
@@ -823,3 +824,58 @@ func (h *Handler) UpdateAdById(c *gin.Context) {
...
+	config, err := h.entities.UpsertUserTag(req)
+	if err != nil {
+		h.log.Error(err, "[handler.UpsertTagme] entity.UpsertTagme() failed")
+		c.JSON(http.StatusInternalServerError, response.CreateResponse[any](nil, nil, err, nil))
+		return
+	}
```
Comment (vhbien000) [inline pkg/handler/community/community.go:849]: should add line break here

File: pkg/handler/community/community.go [hunk 5]
```diff
@@ -823,3 +824,58 @@ func (h *Handler) UpdateAdById(c *gin.Context) {
...
+// @Success     200 {object} response.ResponseMessage
+// @Router      /community/tagme [get]
+func (h *Handler) GetUserTag(c *gin.Context) {
+	userIDStr := c.Query("user_id")
+	userID, _ := strconv.ParseInt(userIDStr, 10, 64)
+	if userID == 0 {
+		h.log.Info("[handler.GetUserTag] missing user_id")
+		c.JSON(http.StatusBadRequest, response.CreateResponse[any](nil, nil, errors.New("user_id is required"), nil))
+		return
+	}
```
Comment (vhbien000) [inline pkg/handler/community/community.go:870]: should add line break here

File: pkg/handler/community/community.go [hunk 6]
```diff
@@ -823,3 +824,58 @@ func (h *Handler) UpdateAdById(c *gin.Context) {
...
+	guildIdStr := c.Query("guild_id")
+	guildID, _ := strconv.ParseInt(guildIdStr, 10, 64)
```
Comment (vhbien000) [inline pkg/handler/community/community.go:872]: why don't we check guild ID (same with above user ID)

File: pkg/handler/community/community.go [hunk 7]
```diff
@@ -823,3 +824,58 @@ func (h *Handler) UpdateAdById(c *gin.Context) {
...
+// @Tags        Community
+// @Accept      json
+// @Produce     json
+// @Param       user_id   query  string true  "User ID"
+// @Param       guild_id   query  string false  "Guild ID"
+// @Success     200 {object} response.ResponseMessage
+// @Router      /community/tagme [get]
+func (h *Handler) GetUserTag(c *gin.Context) {
+	userIDStr := c.Query("user_id")
+	userID, _ := strconv.ParseInt(userIDStr, 10, 64)
```
Comment (vhbien000) [inline pkg/handler/community/community.go:865]: as I remember, query is always a string ,so we should check if the ID is in invalid format/not a number ("ascasc", "*&*hj"...). Do not neet to check if it is zero. 0 is a valid ID, if it is not exist in our DB, another error will be raised when we try to get this from repo.

File: pkg/handler/community/interface.go [hunk 8]
```diff
@@ -35,4 +35,7 @@ type IHandler interface {
 	InitAdSubmission(c *gin.Context)
 	DeleteAdById(c *gin.Context)
 	UpdateAdById(c *gin.Context)
+
```
Comment (vhbien000) [inline pkg/handler/community/interface.go:38]: i think that we do not need this line break, if want to group methods, can add comment above them.

**Review by trkhoi — APPROVED** [review]

**Review by vhbien000 — APPROVED** [review]

File: pkg/handler/community/community.go (hunk 6 above)
Comment (NguyenHuy1812) [reply pkg/handler/community/community.go:872]: guildID in this case is optional query, so not check here

File: migrations/schemas/20230410054010-add-table-user-tags.sql (hunk 1 above)
Comment (NguyenHuy1812) [reply migrations/schemas/20230410054010-add-table-user-tags.sql:5]: actually this table not joined with any table, and Id here just a PK. Not have any special reason, so I will change back to serial id, tks

File: pkg/handler/community/community.go (hunk 6 above)
Comment (vhbien000) [reply pkg/handler/community/community.go:872]: I mean check if it exist, it should be valid, like above when I mentioned about invalid number

**Pull request merged**